starting_package/openvino_workbench/image.py

```diff
@@ -17,150 +17,169 @@
-    if progress_detail and 'current' in progress_detail and 'total' in progress_detail:
-        current = line['progressDetail']['current']
-        total_weight = line['progressDetail']['total']
-
-    # Create progress object
-    progress = {'id': pid, 'status': status,
-                'total_layer_weight': total_weight,
-                'current': current}
-
-    return progress
-
-
-def calculate_total_progress(total_downloaded: int, total_extracted: int, total_image_size: int) -> int:
-    return int(0.5 * (total_downloaded + total_extracted) / total_image_size * 100)
-
-
-def update_progress_bar(progress: int, progress_bar: tqdm):
-    progress_bar.update(progress - progress_bar.n)
-
-
-def parse_image_name(image_name: str) -> tuple:
-    try:
-        repository, tag = image_name.split(':')
-    except ValueError:
-        print(f'The specified image name: "{image_name}" might be incorrect.\n'
-              f'Please specify the image name in the following format: repository:tag\n'
-              f'Example: openvino/workbench:latest\n'
-              f'Or omit the `--image` argument to use the latest version of DL Workbench.\n'
-              f'Aborting.')
-        sys.exit(1)
-    return repository, tag
-
-
-def is_image_present(docker_client: DockerClient, repository_to_search: str, tag_to_search: str) -> bool:
-    if not docker_client.images.list(name=repository_to_search):
-        return False
 
-    for image in docker_client.images.list(name=repository_to_search):
-        if not image.attrs['RepoTags']:
-            continue
-        repository, tag = parse_image_name(image.attrs['RepoTags'][0])
-        if repository == repository_to_search and tag == tag_to_search:
-            return True
-    return False
-
-
-def is_image_present_in_registry(docker_client: DockerClient, image_name: str) -> bool:
-    try:
-        # If the image exists, this will not raise an error and return non-empty object
-        return bool(docker_client.images.get_registry_data(image_name))
-    # Raises error otherwise
-    except Exception:
+class Image:
+    def __init__(self, docker_client: DockerClient, image_name: str, logger: logging.Logger, proxies=None):
+        if proxies is None:
+            proxies = {}
+        self.client = docker_client
+        self.image_name = image_name
+        self.repository, self.tag = self._parse_image_name(self.image_name)
+        self._logger = logger
+        self.proxies = proxies
+        self._is_present = self._is_image_present()
+        self._is_present_in_registry = self._is_image_present_in_registry()
+
+    def pull(self, force_pull: bool = False):
+        self._logger.info(f'Pulling image with the name: {self.image_name}')
+
+        if self._is_present and not force_pull:
+            self._logger.info('Image is present on the machine.')
+            print(f'The specified image: {self.repository}:{self.tag} is present on the machine. Continuing with it...')
```

Comment:
Can we unify logging interface instead of separate using of `logger` and `print`. My proposal is to configure logger  with two handlers (`FileHandler` and `StreamHandler`) and in python modules retrieve two different loggers (via `logging.getLogger(...)`). So, we will be able to do the following:
```
file_logger.info('Image is present on the machine.')
stream_logger.info(f'The specified image: {self.repository}:{self.tag} is present on the machine. Continuing with it...')
```

Reply:
I've decided to use two handlers with one logger - I believe that one logger (`OpenVINO Python Starter Log`) is enough for the starter app. The `INFO` level should suffice for the user visible messages as they (users) can look at the `DEBUG` level messages in the `LOG_FILE`.